pybsn/switch.py

```diff
@@ -0,0 +1,86 @@
+class Switch(object): 
+    """Switch based API operations
+
+    """
+    def __init__(self, client):
+        self.client = client
+
+        # Default configurable values
+        self.name = None
+        self.dpid = None
+        self.fabric_role = None
+        self.leaf_group = None
+
+    @staticmethod
+    def get_switches(client, filter=None):
+
+        if filter:
```

Comment:
I feel like I could make this prettier by having the available filters in a list but I wasn't able to pull it off when I tried.